homeassistant/components/mealie/todo.py

```diff
@@ -0,0 +1,255 @@
+"""Todo platform for Mealie."""
+
+from __future__ import annotations
+
+from aiomealie import MealieError, MutateShoppingItem, ShoppingItem, ShoppingList
+
+from homeassistant.components.todo import (
+    TodoItem,
+    TodoItemStatus,
+    TodoListEntity,
+    TodoListEntityFeature,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .coordinator import MealieConfigEntry, MealieShoppingListCoordinator
+from .entity import MealieEntity
+
+TODO_STATUS_MAP = {
+    False: TodoItemStatus.NEEDS_ACTION,
+    True: TodoItemStatus.COMPLETED,
+}
+TODO_STATUS_MAP_INV = {v: k for k, v in TODO_STATUS_MAP.items()}
+
+
+def _convert_api_item(item: ShoppingItem) -> TodoItem:
+    """Convert Mealie shopping list items into a TodoItem."""
+
+    return TodoItem(
+        summary=item.display,
+        uid=item.item_id,
+        status=TODO_STATUS_MAP.get(
+            item.checked,
+            TodoItemStatus.NEEDS_ACTION,
+        ),
+        due=None,
+        description=None,
+    )
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: MealieConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the calendar platform for entity."""
+    coordinator: MealieShoppingListCoordinator = (
+        entry.runtime_data.shoppinglist_coordinator
+    )
+
+    async_add_entities(
+        MealieShoppingListTodoListEntity(coordinator, shopping_list)
+        for shopping_list in coordinator.shopping_lists
+    )
+
+
+class MealieShoppingListTodoListEntity(MealieEntity, TodoListEntity):
+    """A todo list entity."""
+
+    _attr_supported_features = (
+        TodoListEntityFeature.CREATE_TODO_ITEM
+        | TodoListEntityFeature.UPDATE_TODO_ITEM
+        | TodoListEntityFeature.DELETE_TODO_ITEM
+        | TodoListEntityFeature.MOVE_TODO_ITEM
+    )
+
+    coordinator: MealieShoppingListCoordinator
+
+    def __init__(
+        self, coordinator: MealieShoppingListCoordinator, shopping_list: ShoppingList
+    ) -> None:
+        """Create the Calendar entity."""
+        super().__init__(coordinator, shopping_list.name.lower())
```

Comment:
This is for the unique id right? Can we use the uuid of the shopping list instead?